app/services/topology_service_api.rb

```diff
@@ -0,0 +1,15 @@
+require 'topological_inventory-api-client'
+class TopologyServiceApi
+  attr_accessor :params, :api_instance
+  def initialize(options)
+    @params = options
+    TopologicalInventoryApiClient.configure do |config|
+      # Configure HTTP basic authorization: UserSecurity
+      config.username = 'YOUR USERNAME'
+      config.password = 'YOUR PASSWORD'
+      config.host     = ENV['TOPOLOGY_SERVICE_URL']
```

Comment:
@carbonin 
It can handle scheme://host:port format
https://github.com/swagger-api/swagger-codegen/blob/56fe33b92ee6dcc1d93a1fdea8b5f02cc0ec1d41/samples/client/petstore/ruby/spec/api_client_spec.rb#L12

Reply:
Does this set the scheme and port when you pass a full url? That link only shows that the host is set properly.

If it does set them then you can remove the `config.scheme` line, and if it doesn't you'll need to add a `config.port` line.

Reply:
@carbonin it doesn't set the scheme, the scheme has to be set

Reply:
What about the port?

Reply:
@carbonin 

The host includes the port

I have tested it with localhost:5000 and it works. They are expecting the hostname and port separated by a colon

This is how they set the host, strip out the scheme

https://github.com/mkanoor/topological_inventory-api-client/blob/2ecf303e3211e763d1deff26cb079b91060a7220/lib/topological_inventory-api-client/configuration.rb#L167

When they create the URL they concatenate the scheme, host and basePath

https://github.com/mkanoor/topological_inventory-api-client/blob/2ecf303e3211e763d1deff26cb079b91060a7220/lib/topological_inventory-api-client/configuration.rb#L177

So the host has to include the port number.